src/Keboola/StorageApi/GCSUploader.php

```diff
@@ -235,4 +222,50 @@ private function upload(string $bucket, string $key, array $chunk): void
             }
         }
     }
+
+    private function initClient(array $options): GoogleStorageClient
+    {
+        $this->fetchAuthToken = new class ($options['credentials']) implements FetchAuthTokenInterface {
+            private array $creds;
+
+            public function __construct(
+                array $creds
+            ) {
+                $this->creds = $creds;
+            }
+
+            public function fetchAuthToken(callable $httpHandler = null)
+            {
+                return $this->creds;
+            }
```

Comment:
bylo by zajímavé jestli to neumí interně udělat ten featch credentials tady. Ale co sem sa díval do kódu tak to asi nevolá při 401

Reply:
No premyslal som u toho to hodit sem, skusal som to este debugovat a imho s ato nezavola pri tej 401